frontend/src/metabase/xray/selectors.js

```diff
@@ -21,14 +24,92 @@ export const getSegmentConstituents = (state) =>
         )
     )
 
-export const getComparison = (state) => state.xray.comparison
+export const getComparison = (state) => state.xray.comparison && state.xray.comparison
+
+export const getComparisonFields = createSelector(
+    [getComparison],
+    (comparison) => {
+        if(comparison) {
+            return Object.keys(comparison.constituents[0].constituents)
+                             .map(key => {
+                                 return {
+                                     ...comparison.constituents[0].constituents[key].field,
+                                     distance: comparison.comparison[key].distance
+                                 }
+                             })
+        }
+    }
+)
 
-export const getComparisonFields = (state) => state.xray.comparison && (
-    Object.keys(state.xray.comparison.constituents[0].constituents)
-                     .map(key => {
-                         return {
-                             ...state.xray.comparison.constituents[0].constituents[key].field,
-                             distance: state.xray.comparison.comparison[key].distance
-                         }
-                     })
+export const getComparisonContributors = createSelector(
+    [getComparison],
+    (comparison) => {
+        if(comparison) {
+            const top = comparison['top-contributors']
+            return top && top.map(contributor => {
+                return Object.keys(comparison.constituents[0].constituents)
+                      .map(key => {
+                          if(key === contributor.field.toUpperCase()) {
```

Comment:
I don't completely understand this – this seems to return `undefined` every time the current key doesn't match the contributor field, but those `undefined`s are never filtered out at least in this selector